.github/scripts/boost/build_and_test_boost.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+BOOST_BUILD_DIR=$1
+BOOST_TEMPLATE_DIR=$2
+
+if [[ $# -lt 1 ]]; then
+    echo "Expected at least 1 arguments!" >&2
+    exit 1
+fi
+
+if [[ -d $BOOST_BUILD_DIR ]]; then
+    echo "Directory $BOOST_BUILD_DIR already exists!" >&2
+    exit 1
+fi
+
+mkdir -p $BOOST_BUILD_DIR
+
+if [[ ! -z $BOOST_TEMPLATE_DIR && -d $BOOST_TEMPLATE_DIR ]]; then
+    echo "Found Boost template dir '$BOOST_TEMPLATE_DIR', using it instead of cloning.."
+    time cp -r $BOOST_TEMPLATE_DIR/* $BOOST_BUILD_DIR
+else
+    echo "Cloning Boost GitHub repository.."
+    time git clone --recursive https://github.com/boostorg/boost.git $BOOST_BUILD_DIR
+    # process module contained bug which prevented building it with MinGW, there's already fix
+    # but it's not updated in Boost project yet.
+    cd $BOOST_BUILD_DIR/libs/process
+    git checkout develop
+    cd -
+fi
+
+cd $BOOST_BUILD_DIR
+
+echo "using gcc :  : /home/vejby/cross-aarch64-w64-mingw32-msvcrt/bin/aarch64-w64-mingw32-g++ ;" > "user-config.jam"
+
+echo "Running Boost bootstrap.."
+time ./bootstrap.sh > boost-bootstrap.log
+
+echo "Running Boost b2 build..."
+# --without-context --without-cobalt --without-coroutine --without-fiber
+#    context contains assembler and it doesn't have specified target for MinGW Win ARM64
+#    cobalt, coroutine and fiber depends on context
+#    TODO: Investigate if I just didn't specify args uncorrectly as I had problems with that even
+#       for x64 MinGW compilator
+# --without-charconv --without-json
+#    answer.low = _umul128(a, b, &high); // _umul128 not available on ARM64
+# --without-predef
+#    a library that defines a set of compiler architecture, operating system, library, and other
+#    version numbers it can gather from predefined macros
+#    -> needs to have specified target for MinGW Win ARM 64 as well
+time ./b2 --user-config=./user-config.jam --prefix=./build --debug-configuration target-os=windows address-model=64 variant=debug debug-symbols=on install --without-context --without-cobalt --without-coroutine --without-fiber --without-charconv --without-json --without-predef --without-program_options > boost-build.log
```

Comment:
When the script will be split to the build an test phase, please, remote the time command. Locally it can be specified by the user when executing the script, on the GitHub Actions the time of a job is measured anyway.